lib/prima_auth0_ex/middleware/require_permission.ex

```diff
@@ -0,0 +1,42 @@
+defmodule PrimaAuth0Ex.Middleware.RequirePermission do
+  @moduledoc """
+  Absinthe middleware that ensure the permission is included in the current security context.
+  """
+
+  require Logger
+
+  alias Absinthe.Resolution
+  alias PrimaAuth0Ex.Plug.CreateSecurityContext.Context
+
+  @behaviour Absinthe.Middleware
+
+  @impl true
+  def call(
+        %{context: %Context{dry_run: dry_run, permissions: permissions}} = resolution,
+        required_permission
+      ) do
```

Comment:
Is all this really clearer than a trivial set of [conditions](https://github.com/primait/bburago/blob/master/lib/bburago/graphql/authorization.ex#L66)? 🙂

Reply:
I guess this could be more readable with an hybrid approach, something like
```elixir
if has_required_permission?(permissions, required_permission) do
  resolution
 else
  resolve(dry_run, resolution)
end
```